File: app/src/main/java/com/kaciula/archiman/presentation/screen/userdetails/UserDetailsModule.java
package com.kaciula.archiman.presentation.screen.userdetails;

import com.kaciula.archiman.domain.boundary.SchedulerProvider;
import com.kaciula.archiman.injection.ActivityScope;
import com.kaciula.archiman.presentation.screen.home.UserViewModel;
import dagger.Module;
import dagger.Provides;

@Module
class UserDetailsModule {

  private final UserDetailsContract.View view;
  private UserViewModel user;

  UserDetailsModule(UserDetailsContract.View view, UserViewModel user) {
    this.view = view;
    this.user = user;
  }

  @Provides
  @ActivityScope
  UserDetailsContract.Presenter providePresenter(SchedulerProvider schedulerProvider) {
    return new UserDetailsPresenter(view, user, schedulerProvider);
  }
}

File: app/src/main/java/com/kaciula/archiman/presentation/screens/userdetails/UserDetailsModule.kt
package com.kaciula.archiman.presentation.screens.userdetails

import com.kaciula.archiman.domain.boundary.infrastructure.LocationProvider
import com.kaciula.archiman.domain.util.SchedulerProvider
import com.kaciula.archiman.injection.ActivityScope
import com.kaciula.archiman.presentation.screens.home.UserViewModel
import dagger.Module
import dagger.Provides

@Module
class UserDetailsModule(
    private val view: UserDetailsContract.View,
    private val user: UserViewModel
) {

    @Provides
    @ActivityScope
    fun providePresenter(
        locationProvider: LocationProvider,
        schedulerProvider: SchedulerProvider
    ): UserDetailsContract.Presenter {
        return UserDetailsPresenter(view, user, locationProvider, schedulerProvider)
    }
}
